### PR TITLE
[full-ci] [tests-only] Move tests from `core` to `oCIS` related to `/Shares` of TUS upload

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This PR adds the tests for TUS upload related to /Shares which are removed from the core with this PR https://github.com/owncloud/core/pull/40305 as /Shares related implementation is not in core and is shifted to oCIS.
Bump commit id of core to include PR https://github.com/owncloud/core/pull/40369

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/4154#issuecomment-1206026045
- Part of https://github.com/owncloud/QA/issues/766

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiArchiver-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/15008/42/1) failed. Further test are cancelled...

**Comment by grgprarup** [comment]
Waiting for PR https://github.com/owncloud/ocis/pull/4440 to get merged.

**Comment by saw-jan** [comment]
@grgprarup You have similar PR https://github.com/owncloud/ocis/pull/4540. Please, close one if they are similar.

**Comment by grgprarup** [comment]
Failures in https://drone.owncloud.com/owncloud/ocis/15390/42/8 are due to the space id of the previous scenario step. The space id was not cleared up after its use. After the merge of the PR https://github.com/owncloud/core/pull/40369, this PR will pass on CI.

**Review by SagarGi — COMMENTED** [review]
long but LGTM :+1:

File: tests/acceptance/features/bootstrap/SpacesTUSContext.php [hunk 1]
```diff
@@ -117,8 +118,8 @@ public function userUploadsAFileViaTusInsideOfTheSpaceUsingTheWebdavApi(
 	 * @throws GuzzleException
 	 */
 	public function userHasCreatedANewTusResourceForTheSpaceUsingTheWebdavApiWithTheseHeaders(
-		string $user,
-		string $spaceName,
+		string    $user,
+		string    $spaceName,
```
Comment (amrita-shrestha) [inline tests/acceptance/features/bootstrap/SpacesTUSContext.php:122]:
```suggestion
		string $user,
		string $spaceName,
```
why adding a tab?

File: tests/acceptance/features/bootstrap/SpacesTUSContext.php [hunk 2]
```diff
@@ -138,8 +139,8 @@ public function userHasCreatedANewTusResourceForTheSpaceUsingTheWebdavApiWithThe
 	 * @throws GuzzleException
 	 */
 	public function userCreatesANewTusResourceForTheSpaceUsingTheWebdavApiWithTheseHeaders(
-		string $user,
-		string $spaceName,
+		string    $user,
+		string    $spaceName,
```
Comment (amrita-shrestha) [inline tests/acceptance/features/bootstrap/SpacesTUSContext.php:143]:
```suggestion
		string $user,
		string $spaceName,
```

File: tests/acceptance/features/bootstrap/SpacesTUSContext.php [hunk 3]
```diff
@@ -166,4 +167,113 @@ public function userUploadsAFileWithContentToViaTusInsideOfTheSpaceUsingTheWebda
...
+	 * @param string $destination
+	 * @param string $mtime Time in human readable format is taken as input which is converted into milliseconds that is used by API
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 *
+	 * @throws Exception
+	 * @throws GuzzleException
+	 */
+	public function userUploadsAFileToWithMtimeViaTusInsideOfTheSpaceUsingTheWebdavApi(string $user, string $source, string $destination, string $mtime, string $spaceName): void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesTUSContext.php:185]: too long lines. good to break them
```suggestion
	public function userUploadsAFileToWithMtimeViaTusInsideOfTheSpaceUsingTheWebdavApi(
	        string $user, 
	        string $source,
	        string $destination,
	        string $mtime,
	        string $spaceName
	): void {
```

File: tests/acceptance/features/apiSpaces/shareUploadTUS.feature [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+    And user "Alice" has shared folder "/toShare" with user "Brian"
+    And user "Brian" has accepted share "/toShare" offered by user "Alice"
+    When user "Brian" uploads a file "filesForUpload/textfile.txt" to "toShare/file.txt" with mtime "Thu, 08 Aug 2012 04:18:13 GMT" via TUS inside of the space "Shares Jail" using the WebDAV API
+    Then for user "Brian" folder "toShare" of the space "Shares Jail" should contain these entries:
+      | file.txt |
+    And as "Brian" the mtime of the file "/toShare/file.txt" in space "Shares Jail" should be "Thu, 08 Aug 2012 04:18:13 GMT"
+    And as "Alice" the mtime of the file "/toShare/file.txt" in space "Personal" should be "Thu, 08 Aug 2012 04:18:13 GMT"
+
+
+  Scenario: upload file with mtime to a send share
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/shareUploadTUS.feature:26]:
```suggestion
  Scenario: upload file with mtime to a sent share
```

File: tests/acceptance/features/apiSpaces/shareUploadTUS.feature [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+    And user "Brian" has accepted share "/toShare" offered by user "Alice"
+    And user "Alice" has uploaded file with content "uploaded content" to "/toShare/file.txt"
+    When user "Brian" uploads a file "filesForUpload/textfile.txt" to "toShare/file.txt" with mtime "Thu, 08 Aug 2012 04:18:13 GMT" via TUS inside of the space "Shares Jail" using the WebDAV API
+    Then for user "Brian" folder "toShare" of the space "Shares Jail" should contain these entries:
+      | file.txt |
+    And as "Brian" the mtime of the file "/toShare/file.txt" in space "Shares Jail" should be "Thu, 08 Aug 2012 04:18:13 GMT"
+    And as "Alice" the mtime of the file "/toShare/file.txt" in space "Personal" should be "Thu, 08 Aug 2012 04:18:13 GMT"
+
+
+  Scenario: overwriting a file with mtime in a send share
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/shareUploadTUS.feature:49]:
```suggestion
  Scenario: overwriting a file with mtime in a sent share
```

File: tests/acceptance/features/apiSpaces/shareUploadTUS.feature [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+    Given using OCS API version "1"
+    And user "Alice" has created folder "/toShare"
+    And user "Alice" has shared folder "/toShare" with user "Brian"
+    And user "Brian" has accepted share "/toShare" offered by user "Alice"
+    When user "Brian" uploads a file with content "uploaded content" to "/toShare/nonExistentFolder/file.txt" via TUS inside of the space "Shares Jail" using the WebDAV API
+    Then for user "Brian" folder "toShare" of the space "Shares Jail" should not contain these entries:
+      | nonExistentFolder |
+    And for user "Brian" folder "toShare" of the space "Shares Jail" should not contain these entries:
+      | nonExistentFolder/file.txt |
+
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/shareUploadTUS.feature:71]: I think this check is enough and this is not needed
```
 Then for user "Brian" folder "toShare" of the space "Shares Jail" should not contain these entries:
      | nonExistentFolder |
```

File: tests/acceptance/features/apiSpaces/shareUploadTUS.feature [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+  Scenario: attempt to upload a file into a nonexistent folder within correctly received read only share
+    Given using OCS API version "1"
+    And user "Alice" has created folder "/toShare"
+    And user "Alice" has shared folder "/toShare" with user "Brian" with permissions "read"
+    And user "Brian" has accepted share "/toShare" offered by user "Alice"
+    When user "Brian" uploads a file with content "uploaded content" to "/toShare/nonExistentFolder/file.txt" via TUS inside of the space "Shares Jail" using the WebDAV API
+    Then for user "Brian" folder "toShare" of the space "Shares Jail" should not contain these entries:
+      | nonExistentFolder |
+    And for user "Brian" folder "toShare" of the space "Shares Jail" should not contain these entries:
+      | nonExistentFolder/file.txt |
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/shareUploadTUS.feature:82]: same here

File: tests/acceptance/features/apiSpaces/shareUploadTUS.feature [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+    Given user "Alice" has created folder "/toShare"
+    And user "Alice" has shared folder "/toShare" with user "Brian"
+    And user "Brian" has accepted share "/toShare" offered by user "Alice"
+    When user "Brian" uploads a file with content "uploaded content" to "/toShare/file.txt" via TUS inside of the space "Shares Jail" using the WebDAV API
+    Then for user "Alice" folder "toShare" of the space "Personal" should contain these entries:
+      | file.txt |
+    And for user "Alice" the content of the file "toShare/file.txt" of the space "Personal" should be "uploaded content"
+
+
+  Scenario: Uploading file to a user read/write share folder works
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/shareUploadTUS.feature:95]:
```suggestion
  Scenario: Uploading a file to a user read/write share folder
```

File: tests/acceptance/features/apiSpaces/shareUploadTUS.feature [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+    And user "Alice" has shared folder "/toShare" with user "Brian" with permissions "read"
+    And user "Brian" has accepted share "/toShare" offered by user "Alice"
+    When user "Brian" uploads a file with content "uploaded content" to "/toShare/nonExistentFolder/file.txt" via TUS inside of the space "Shares Jail" using the WebDAV API
+    Then for user "Brian" folder "toShare" of the space "Shares Jail" should not contain these entries:
+      | nonExistentFolder |
+    And for user "Brian" folder "toShare" of the space "Shares Jail" should not contain these entries:
+      | nonExistentFolder/file.txt |
+
+
+  Scenario: Uploading file to a received share folder
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/shareUploadTUS.feature:85]:
```suggestion
  Scenario: Uploading a file to a received share folder
```

File: tests/acceptance/features/apiSpaces/shareUploadTUS.feature [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+    Given user "Alice" has created folder "/toShare"
+    And user "Alice" has shared folder "/toShare" with user "Brian" with permissions "change"
+    And user "Brian" has accepted share "/toShare" offered by user "Alice"
+    When user "Brian" uploads a file with content "uploaded content" to "/toShare/file.txt" via TUS inside of the space "Shares Jail" using the WebDAV API
+    Then for user "Alice" folder "toShare" of the space "Personal" should contain these entries:
+      | file.txt |
+    And for user "Alice" the content of the file "toShare/file.txt" of the space "Personal" should be "uploaded content"
+
+
+  Scenario: Uploading a file into a group share as share receiver
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/shareUploadTUS.feature:105]:
```suggestion
  Scenario: Uploading a file into a group share as a share receiver
```

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Review by amrita-shrestha — COMMENTED** [review]
lgtm

**Review by phil-davis — APPROVED** [review]

**Pull request merged**